src/platforms/stlinkv3/platform.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2011-2021 Black Sphere Technologies Ltd.
+ * Portions (C) 2020-2021 Stoyan Shopov <stoyan.shopov@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* This file implements the platform specific functions for the STM32
+ * implementation.
+ */
+#ifndef PLATFORMS_STLINKV3_PLATFORM_H
+#define PLATFORMS_STLINKV3_PLATFORM_H
+
+#include "gpio.h"
+#include "timing.h"
+#include "timing_stm32.h"
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/stm32/f1/memorymap.h>
+#include <libopencm3/usb/usbd.h>
+
+#ifdef ENABLE_DEBUG
+#define PLATFORM_HAS_DEBUG
+extern bool debug_bmp;
+#endif
+
+#define PLATFORM_HAS_USBUART
+#define PLATFORM_IDENT "STLINK-V3 "
+
+#define BOOTMAGIC0 0xb007da7a
+#define BOOTMAGIC1 0xbaadfeed
+
+#define DESIG_UNIQUE_ID_BASE DESIG_UNIQUE_ID_BASE_452
+
+/* Hardware definitions... */
+#define TDI_PORT GPIOA
+#define TMS_PORT GPIOF
+#define TCK_PORT GPIOH
+#define TDO_PORT GPIOD
+#define TDI_PIN  GPIO1
+#define TMS_PIN  GPIO9
+#define TCK_PIN  GPIO6
+#define TDO_PIN  GPIO2
+
+#define SWDIO_IN_PORT GPIOH
+#define SWDIO_PORT    TMS_PORT
+#define SWCLK_PORT    TCK_PORT
+#define SWDIO_IN_PIN  GPIO7
+#define SWDIO_PIN     TMS_PIN
+#define SWCLK_PIN     TCK_PIN
+
+#define SRST_PORT GPIOA
+#define SRST_PIN  GPIO6
+
+#define TMS_DRIVE_PORT GPIOA
+#define TMS_DRIVE_PIN  GPIO7
+
+#define PLATFORM_HAS_TRACESWO 1
+#define NUM_TRACE_PACKETS     (16)
+#define TRACESWO_PROTOCOL     2 /* 1 = Manchester, 2 = NRZ / async */
+
+#define SWDIO_MODER      GPIO_MODER(TMS_PORT)
+#define SWDIO_MODER_MULT (1 << (9 << 1))
+
+#define TMS_SET_MODE()                                                    \
+	gpio_mode_setup(TMS_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, TMS_PIN); \
+	gpio_set_output_options(TMS_PORT, GPIO_OTYPE_PP, GPIO_OSPEED_2MHZ, TMS_PIN);
+
+#define SWDIO_MODE_FLOAT()                         \
+	do {                                           \
+		uint32_t moder = SWDIO_MODER;              \
```

Comment:
`moder` looks like a typo for `mode`, leading to a bit of  head-scratching reading this. Let's correct the naming?

On second reading, it's likely the `r` is supposed to mean `register`, so for the `#define` above (`SWDIO_MODER`), it probably makes sense to rename to `SWDIO_MODE_REG` to make that obvious